core/src/gravatar.rs

```diff
@@ -0,0 +1,87 @@
+// check-if-email-exists
+// Copyright (C) 2018-2022 Reacher
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use crate::util::constants::LOG_TARGET;
+use md5;
+use md5::Digest;
+use serde::{Deserialize, Serialize};
+
+const API_BASE_URL: &str = "https://www.gravatar.com/avatar/";
+
+// Details on whether the given email address has a gravatar image.
+#[derive(Debug, Deserialize, Serialize)]
+pub struct GravatarDetails {
+	// Whether a gravatar image for the given email exists.
+	pub has_image: bool,
+	// The Gravatar url of the image belonging to the given email.
+	pub url: Option<String>,
+}
+
+impl Default for GravatarDetails {
+	fn default() -> Self {
+		GravatarDetails {
+			has_image: false,
+			url: None,
+		}
+	}
+}
+
+pub async fn check_gravatar(to_email: &str) -> GravatarDetails {
+	let client = reqwest::Client::new();
```

Comment:
This might make each email verification longer. Is it possible to add a `check_gravater: bool` field in CheckEmailInput, defaulted to false, so that the gravatar check is only checked when that bool is true?